examples/other/integration_pytorch.py

```diff
@@ -0,0 +1,694 @@
+    def _create_background(top_color: gfx.Color, bottom_color: gfx.Color) -> gfx.Background:
+        return gfx.Background(geometry=None, material=gfx.BackgroundMaterial(top_color, bottom_color))
+
+    @staticmethod
+    def _create_background_scene(renderer: gfx.Renderer, color: gfx.Color) -> Tuple[gfx.Viewport, gfx.Camera, gfx.Background]:
+        viewport = gfx.Viewport(renderer)
+        camera = gfx.NDCCamera()
+        background = SceneHandler._create_background(top_color=color, bottom_color=color)
+        return viewport, camera, background
+
+    @staticmethod
+    def _create_scene(renderer: gfx.Renderer, light_color: gfx.Color, background_top_color: gfx.Color, background_bottom_color: gfx.Color, rect_length: int = 1) -> Tuple[gfx.Viewport, gfx.Camera, gfx.Scene]:
+        viewport = gfx.Viewport(renderer)
+        camera = gfx.PerspectiveCamera()
+        camera.show_rect(left=-rect_length, right=rect_length, top=-rect_length, bottom=rect_length, view_dir=(-1, -1, -1), up=(0, 0, 1))
+        controller = gfx.OrbitController(camera=camera, register_events=viewport)
+        light = gfx.PointLight(color=light_color, intensity=5, decay=0)
+        background = SceneHandler._create_background(top_color=background_top_color, bottom_color=background_bottom_color)
+        scene = gfx.Scene()
+        scene.add(light)
+        scene.add(background)
+        return viewport, camera, scene
+
+    @staticmethod
+    def _create_text_scene(text_string: str, position: np.ndarray, anchor: str, font_size: int) -> Tuple[gfx.Scene, gfx.Camera]:
+        scene = gfx.Scene()
+        camera = gfx.ScreenCoordsCamera()
+        text = SceneHandler._create_world_object_for_text(
+            text_string=text_string,
+            anchor=anchor,
+            position=position,
+            font_size=font_size)
+        scene.add(text)
+        return scene, camera
+
+    @staticmethod
+    def _animate():
+        try:
+            mesh_data = cast(MeshData, SceneHandler._in_queue.get_nowait())
+
+            # if that's the first mesh-data message, create meshes
+            if SceneHandler._scene_state.first_data:
+                SceneHandler._scene_state.gt_mesh = SceneHandler._create_world_object_for_mesh(
+                    mesh_data=mesh_data,
+                    k=mesh_data.k)
+                SceneHandler._scene_state.pred_mesh = SceneHandler._create_world_object_for_mesh(
+                    mesh_data=mesh_data,
+                    k=mesh_data.pred_k)
+                SceneHandler._scene_state.gt_group.add(SceneHandler._scene_state.gt_mesh)
+                SceneHandler._scene_state.pred_group.add(SceneHandler._scene_state.pred_mesh)
+                SceneHandler._scene_state.first_data = False
+            # otherwise, update mesh colors
+            else:
+                c_pred = SceneHandler._get_vertex_colors_from_k(k=mesh_data.pred_k)
+                SceneHandler._scene_state.pred_mesh.geometry.colors = gfx.Buffer(data=c_pred)
```

Comment:
Is `c_pred` always the same size? If it is, it's likely more efficient to do `geometry.color.data[:] = c_pred` and then `geometry.colors.update_range()`.

Reply:
Fixed